### PR TITLE
feat: rewrite the preset

### PR DESCRIPTION
**Pull request by awwwdev**
This is a complete rewrite of the preset and many missing features are added.

Instead of adding all 12 shades of a color at the start, this adds color shades as they are used, unless they are safeListed. So always, the minimum amount of css is added to your project.

Mechanism:
There is a colorsInUse object that keeps track of colors used. 
There is a unocss shortcut that runs when a radix color is used in a utility token, and add the color to colorsInUse.
In preflight, based on colorsInUse, it add css variables.

palette option is removed.

new options are introduced.

useP3Colors (boolean) 
to use P3 colors with fallbacks

safeListColors: 
a list of colors or color-shades to preserve

safeListAliases:
a list of aliases to preserve

onlyOneTheme ('dark' | 'light' | undefined):
if your app has only one theme, you can indicate with this, so only 'dark' or 'light' colors are added to css.

### CODE REVIEW

**Review by awwwdev — COMMENTED** [review]
rewrite

**Comment by endigma** [comment]
Hey, I'd be willing to review and accept this, but you'd need to update docs to reflect your changes and increment the major version in `package.json`.

**Comment by awwwdev** [comment]
Sorry for late response. I'll address those. Thanks.

**Comment by awwwdev** [comment]
Hello again 👋

So I refactored it, fixed the Typescript errors, and wrote docs for changed features, plus a migrations guide from v2.

Please review and let me know what you think. 
I'm open to discuss and change things. 

Thanks

**Comment by endigma** [comment]
Hey, I understand I'm being annoying here but could you maybe `git rebase --fixup` all the repetitive commits and maybe set up SSH or GPG commit signing? I'll review the actual changes soon.

**Comment by awwwdev** [comment]
Hey @endigma ! Not at all. I learned things!
I added ssh commit signing.
Didn't know about commit --fixup.
I will use it later on. But for fixing, I rebased the branch to squash the commits with same message to one. Sorry if it's not clean. I'm new with this. 

Note, if you follow commits the changes on docs might confuse you. I tried to move current doc to a V2 folder and add my doc to V3 folder, but couldn't figure out how to handle things (like in colors page, v2 use radix-colors v2 and v3 radix-colors v3, I didn't know how to use two version of a radix-colors in same time). So, eventually, I gave up and overwrite your docs (v2) with my docs (v3).

Let me know if there is anything confusion, or any suggestion you have.

**Comment by awwwdev** [comment]
@endigma I just wanted to know if you had time to check the pull request.

File: .vscode/settings.json
Comment (endigma) [inline .vscode/settings.json:1]: Please don't add configuration for extensions into the workspace.

File: docs/.vitepress/config.mts [hunk 1]
```diff
@@ -1,58 +1,65 @@
...
-            darkSelector: ".dark",
-
+            darkSelector: '.dark',
+            lightSelector: ':root, .light',
             aliases: {
-              testa: "red",
-              testb: "amber",
+              danger: 'red',
+              warning: 'amber',
+              accent: 'blue',
```
Comment (endigma) [inline docs/.vitepress/config.mts:22]: This is a common thing, not just these lines: please configure whatever formatter you're using to not change every double quote in the project to a single quote, this is not the project style.

File: docs/.vitepress/config.mts [hunk 2]
```diff
@@ -1,58 +1,65 @@
...
-          ...radixColors.map((p) => `hue-${p}`),
-          ...Array.from({ length: 12 }, (_, i) => `bg-hue${i + 1}`),
-          ...Array.from({ length: 12 }, (_, i) => `bg-hue${i + 1}A`),
+          ...['', 'A']
+            .map((alpha) =>
+              [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12].map((step) => RADIX_HUES.map((hue) => `bg-${hue}${step}${alpha}`))
+            )
+            .flat()
+            .flat(),
+          ...RADIX_HUES.map((h) => [`text-${h}9`, `text-${h}-fg`,  `text-${h}12`,  `text-${h}1`]).flat(),
```
Comment (endigma) [inline docs/.vitepress/config.mts:34]: Pretty sure if you reverse the order of these two maps you can get rid of the `flat()` calls. (colors -> alphas)

File: docs/.vitepress/config.mts.timestamp-1728718892247-847c41b0abac5.mjs
Comment (endigma) [inline docs/.vitepress/config.mts.timestamp-1728718892247-847c41b0abac5.mjs:1]: This file does not belong here.

File: docs/v3/migrating-from-v2.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Migrating from v2
+
+## Breaking changes
+
+- V3 uses Radix Colors V3 under the hood. Radix Colors are changed slightly in V3. Please note in V3, step 6, 7 and 8 are used for borders. (Steps 5,6 and 7 were used for borders in Radix Colors V2). Please visit [Radix Docs](https://www.radix-ui.com/colors/docs/palette-composition/understanding-the-scale) for more information.
+
+- `palette` option is removed. You don't need to specify palette colors anymore. Colors are added automatically and only the minimum required CSS variables are added to keep the CSS optimized. But if you need to safelist colors or aliases, use `safelist` option instead. Learn more about safelist option in [Configuration docs](/v3/configuration#safelist) section.
+
+- `hue` utility is removed in favor of more flexible Alias Utility. You can replace `hue-red` and `bg-hue4` with `alias-hue-red` and use`bg-hue4`. You can use any other aliases instead of `hue`.
```
Comment (endigma) [inline docs/v3/migrating-from-v2.md:9]: Missing a space between use and `

File: src/validation.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+export function isValidAliasName(aliasName: string) {
+  if (typeof aliasName !== 'string') return false
+  if (['white', 'black', ...RADIX_HUES].includes(aliasName)) return false;
+  if (!aliasName.match(/^[a-z]+(-[a-z]+)*$/)) return false;
+  return true;
+}
+
+export function isValidRadixHue(hue: string) {
+  // @ts-ignore
```
Comment (endigma) [inline src/validation.ts:19]: Please do not ignore typescript errors

**Review by endigma — CHANGES_REQUESTED** [review]
a first review pass, don't have time to read all the docs copy right now

File: src/validation.ts (hunk 4 above)
Comment (awwwdev) [reply src/validation.ts]: fixed

**Comment by awwwdev** [comment]
@endigma Thank you for your review. All issues are fixed.

**Comment by endigma** [comment]
You should fork the project and release on npm under your own name if you want this to be public. I've had a more thorough look and while I don't like how the current code is, at least its _my_ code I don't like. I don't want to accept your full rewrite under my repo and effectively be maintaining your code, including any issues it may cause existing users.

**Comment by awwwdev** [comment]
This option is easier for me too, but I wanted to avoid duplicate packages, so people using this will benefit from updates, and also try to work with other people on OSS development. At least I tried.
As a developer I understand your point. But, I wish you could express this earlier.

**Pull request closed without merging**